base/commands/migration/status_stages_it_test.go

```diff
@@ -0,0 +1,127 @@
+//go:build std || migration
+
+package migration_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	_ "github.com/hazelcast/hazelcast-commandline-client/base"
+	_ "github.com/hazelcast/hazelcast-commandline-client/base/commands"
+	"github.com/hazelcast/hazelcast-commandline-client/base/commands/migration"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+	hz "github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStatus(t *testing.T) {
+	testCases := []struct {
+		name string
+		f    func(t *testing.T)
+	}{
+		{name: "status", f: statusTest},
+		{name: "noMigrationsStatus", f: noMigrationsStatusTest},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, tc.f)
+	}
+}
+
+func noMigrationsStatusTest(t *testing.T) {
+	tcx := it.TestContext{T: t}
+	ctx := context.Background()
+	tcx.Tester(func(tcx it.TestContext) {
+		var wg sync.WaitGroup
+		wg.Add(1)
+		var execErr error
+		go tcx.WithReset(func() {
+			defer wg.Done()
+			execErr = tcx.CLC().Execute(ctx, "status")
+		})
+		wg.Wait()
+		require.Contains(t, execErr.Error(), "there are no migrations are in progress on migration cluster")
+	})
+}
+
+func statusTest(t *testing.T) {
+	tcx := it.TestContext{T: t}
+	ctx := context.Background()
+	tcx.Tester(func(tcx it.TestContext) {
+		ci := hz.NewClientInternal(tcx.Client)
+		mID := preStatusRunner(t, tcx, ctx, ci)
+		defer removeMembersLogs(ctx, ci)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go tcx.WithReset(func() {
+			defer wg.Done()
+			Must(tcx.CLC().Execute(ctx, "status"))
+		})
+		// statusRunner removes __datamigrations_in_progress list, so we should give some time to command to read it first
+		time.Sleep(1 * time.Second)
```

Comment:
Sleeps may cause flaky tests.
Let's use `it.Eventually` instead to wait for the condition necessary to proceed.

Reply:
There is no condition to wait in here, so I cannot use `Eventually`.